### DISCUSSION TITLE
MongoDB | string instead of ObjectId (other document in another collection)

### DISCUSSION THREAD

**Discussion by Opus26** [Budibase/budibase#7375] [Help] [answered]
Hello, 

When trying to work with multiple ObjectIds, only "_id" displays as the correct type in MongoDB.

Here is an **example**:
![image](https://user-images.githubusercontent.com/71282839/185718062-4378e0b0-89c6-4007-9866-42c03f916e5a.png)

The **expected result** would be the following:
![image](https://user-images.githubusercontent.com/71282839/185717896-08a028bc-643c-432b-b9c3-b0aec61d49ab.png)

Here is the result I get **instead**:
![image](https://user-images.githubusercontent.com/71282839/185717853-7bec1611-0104-4899-8cb8-617c17f202c9.png)

Is this behavior normal? It would be useful for me to create an ObjectId for the itemId attribute, how can I get around this problem?

Thanks you :)

**Comment by melohagan** [comment, marked as answer]
Hey @Opus26 

Thanks for raising this! The current code doesn't account for relationship fields. 
The current problem of using JSON is that there's no way to know if a user actually wants to query an object id or the string of "ObjectId".
It probably is safe to assume 99.99% of the time they'll want an object id, in which case I can get a fix out for that in the next release.

The object ids will definitely support `$` operators in next release: https://github.com/Budibase/budibase/pull/6895/files. 
So you'd be able to do this as well in future.
```json
"itemId": {
  "$eq": "ObjectId('{{id}}')"
}
```
